File: app-java/src/main/java/net/mEmoZz/yts/java/ui/detail/DetailContract.java
package net.mEmoZz.yts.java.ui.detail;

import android.app.Activity;
import android.content.Context;
import net.mEmoZz.yts.java.data.Quality;
import net.mEmoZz.yts.java.ui.base.BasePresenter;
import net.mEmoZz.yts.java.ui.base.BaseView;
import net.mEmoZz.yts.java.ui.detail.adapters.pager.ScreenshotsAdapter;
import net.mEmoZz.yts.java.ui.detail.adapters.recycler.CastAdapter;

/**
 * Authored by Mohamed Fathy on 10 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

public interface DetailContract {

  interface View extends BaseView<Presenter> {

    void receiveData();

    void setupRecycler();

    void initDialog();

    void loadDetails();

    void loadImages(String tubeUrl, String posterUrl, String backgroundUrl);

    void setInfo(String year, String genre, String rate, String description);

    void setPagerAdapter(ScreenshotsAdapter adapter);

    void setCastRecyclerAdapter(CastAdapter adapter);

    void showMainView();

    void showYear();

    void hideYear();

    void showGenres();

    void hideGenres();

    void showRate();

    void hideRate();

    void enable3D();

    void enable720p();

    void enable1080p();

    void hideSynopsis();

    void hideScreenshots();

    void hideCast();

    void showProgressDialog();

    void dismissProgressDialog();

    void showFileSavedToast();

    void showFileNotSavedToast();

    void showCopyConfirmToast();

    void showNoConnectionToast();

    void showPermissionNotGrantedToast();
  }

  interface Presenter extends BasePresenter<View, DetailInteractor> {

    void loadMovieDetails(Activity context, long movieId);

    void onPermGranted(Context context, Quality quality, boolean isGranted);

    void copyMagnet(Context context, Quality quality);

    String getMovieSize(Quality quality);
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/detail/DetailContract.kt
package net.mEmoZz.yts.kotlin.ui.detail

import android.app.Activity
import android.content.Context
import net.mEmoZz.yts.kotlin.data.Quality
import net.mEmoZz.yts.kotlin.ui.base.BasePresenter
import net.mEmoZz.yts.kotlin.ui.base.BaseView
import net.mEmoZz.yts.kotlin.ui.detail.adapters.pager.ScreenshotsAdapter
import net.mEmoZz.yts.kotlin.ui.detail.adapters.recycler.CastAdapter

/**
 * Authored by Mohamed Fathy on 10 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

interface DetailContract {

  interface View : BaseView<Presenter> {

    fun setupRecycler()

    fun initDialog()

    fun loadImages(tubeUrl: String, posterUrl: String, backgroundUrl: String)

    fun setInfo(year: String, genre: String, rate: String, description: String)

    fun setPagerAdapter(adapter: ScreenshotsAdapter)

    fun setCastRecyclerAdapter(adapter: CastAdapter)

    fun showMainView()

    fun showYear()

    fun hideYear()

    fun showGenres()

    fun hideGenres()

    fun showRate()

    fun hideRate()

    fun enable3D()

    fun enable720p()

    fun enable1080p()

    fun hideSynopsis()

    fun hideScreenshots()

    fun hideCast()

    fun showProgressDialog()

    fun dismissProgressDialog()

    fun showFileSavedToast()

    fun showFileNotSavedToast()

    fun showCopyConfirmToast()

    fun showNoConnectionToast()

    fun showPermissionNotGrantedToast()

    fun initClicks()
  }

  interface Presenter : BasePresenter<View, DetailInteractor> {

    fun loadMovieDetails(context: Activity, movieId: Long)

    fun onPermGranted(context: Context, quality: Quality, isGranted: Boolean)

    fun copyMagnet(context: Context, quality: Quality)

    fun getMovieSize(quality: Quality): String?
  }
}
